docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,138 @@
+1. docker volume inspect - the volume must be qualified by the remote dir label as below,
+   1. _docker volume inspect vol-name@remote-dir-label_
+1. docker run - the volume must be qualified by the remote dir label as below,
+   1. _docker run -it --volume-driver=vsphere -v remote-vol@remote-dir-label:/vol1 --name ub ubuntu_
+1. docker volume rm - the volume must be qualified by the remote dir label as below,
+   1. _dovker volume rm vol-name@remote-dir-label_
+
+
+# Extending the vSphere driver to support shared volumes
+
+The vSphere plugin driver supports an extendable framework that allows adding support for volumes backed by specific storage types. Presently the driver supports VMDK backed volumes on storage available to the local ESX host. A new module to support file backed volumes (on local or NFS mounted directories) is added to the existing framework that supports all of the docker volume functions as listed above.
+
+The new file backed volumes will support NFS v3/v4.0 protocols to mount and use remote directories to host docker volumes. Docker volumes will be supported as folders on the mounted directories and exported to containers in the same way that VMDK volumes are made available to containers.
+
+# Plugin configuration for remote directories
+
+The plugin configuration defined in /etc/docker-volume-vsphere.conf must be modified to include the details of the remote directories to use to locate volumes. The format describing a remote directory is as shown below,
+
+## /etc/docker-volume-vsphere.conf
+{
+"MaxLogAgeDays": 28,
+"MaxLogSizeMb": 100,
+"LogPath": "/var/log/docker-volume-vsphere.log",
+"LogLevel": "debug",
+_“RemoteDirs”: {_
+     _“<dir-A-label>” :   {_
+          _“addr”: <IP>,_
+          _“args: <mount args>,_
+          _“path”: <remote path to mount>,_
+          _“type”: <FS type – nfs3, nfs4, efs, cifs, etc>,_
+          _“vol-path”: <optional path name under path above to locate volumes>_
+	  _"src": <label of another entry in this list to use to locate volumes>_
+	  _"default": <0/1>_
+     _},_
+     _“<dir-B-label>” :   {_
+          _“addr”: <IP>,_
+          _“args: <mount args>,_
+          _“path”: <remote path to mount>,_
+          _“type”: <FS type – nfs3, nfs4, efs, cifs, etc>,_
+          _“vol-path”: <optional path name under path above to locate volumes>_
+	  _"src": <label of another entry in this list to use to locate volumes>_
+	  _"default": <0/1>_
+     _},_
+     _},_
+     _“<dir-C-label>” :   {_
+          _“addr”: "",_
+          _“args: "",_
+          _“path”: "",_
+          _“type”: "",_
+          _“vol-path”: <optional path name under path above to locate volumes>_
+	  _"src": "dir-A-label"_
+	  _"default": <0/1>_
+     _},_
+     _},_
+  _}_
+}
+
+The configuration items are described below:
+
+1. “RemoteDirs”:  Map of remote dirs.
+1. “<dir-A-label>” – must be unique and user defined and identifies a specific remote dir or another entry within the list
```

Comment:
No, the first create/mount/remove/list/get will mount the remote dir and if necessary create the dir. path. Its possible user just does docker volume ls with a remote dir spec in the config file.